addons/stock/models/stock_quant.py

```diff
@@ -412,18 +412,25 @@ def _update_available_quantity(self, product_id, location_id, quantity, lot_id=N
             try:
                 with self._cr.savepoint():
                     self._cr.execute("SELECT 1 FROM stock_quant WHERE id = %s FOR UPDATE NOWAIT", [quant.id], log_exceptions=False)
+                    new_quantity = quant.quantity + quantity
```

Comment:
As discussed with @amoyaux, running this logic here can introduce slowness. We are thinking it's better to leave the scheduler part, remove this one in stock_quant.py and maybe call run run_inventory_task() before starting any inventory to add and remove products to correct